### ISSUE TITLE
Forward mode results in "PARTICLE_STILT.DAT does not contain any trajectory data" while backward mode works

### ISSUE THREAD

**Issue by BlueGrisGris** [uataq/stilt#99]
Hello,

When I try to run STILT in forward mode with HRRR ARL's from `ftp://ftp.arl.noaa.gov/archives/hrrr/` I have issues with forward in time simulations but not backward in time simulations.  I have double checked that the necessary met files are in the met directory.  The issue seems to arise when trying to cross HRRR file boundaries.  If `run_time` is at 10:00 UTC, it exhibits the failing behavior.  But if run_time is at 12:00 UTC, it works.   This behavior is not entirely consistent however, and I cannot figure out what controls it.  gfs0p25 ARL's seem to work fine.

Thanks!
 

Backward in time:
```
n_hours   <- -6 
```
Results in successful trajectory calculations:
 ```
R
-----------------------------------------
Initializing STILT
Commit ID: fc4daceee2f608650040688213fdb53554d37b62
Number of receptors: 1
Number of parallel workers: 1
Parallelization disabled. Executing simulations sequentially...
Running simulation ID:   202208061000_-103.5_31.5_10
>
```
Trying to go forward with the same meteorology, receptors, only changing `n_hours`:
```
n_hours   <- 6 
```
results in: 
 ```
R
-----------------------------------------
Initializing STILT
Commit ID: fc4daceee2f608650040688213fdb53554d37b62
Number of receptors: 1
Number of parallel workers: 1
Parallelization disabled. Executing simulations sequentially...
Running simulation ID:   202208061000_-103.5_31.5_10
Warning message:
In calc_trajectory(namelist, rundir, emisshrs, hnf_plume, met_files,  
/n/home00/emanninen/apps/stilt/myproject/out/by-id/202208061000_-103.5_31.5_10
PARTICLE_STILT.DAT does not contain any trajectory data.
```
Examining stilt.log reveals: 
```
stilt.log
--------------------------------------------
STOP 900
 HYSPLIT - Initialization
 HYSPLIT version: hysplit.v5.1.0
 Last Changed Date: 2021-05-13
  NOTICE   main: using namelist file - SETUP.CFG
  NOTICE   main: STILT emulation (mixing ratio, vary layer, mass sum)
 Calculation Started ... please be patient
 *ERROR* metpos: start point not within (x,y,t) any data file
  - start time before start of meteorology data
/n/home00/emanninen/apps/stilt/myproject/out/by-id/202208061000_-103.5_31.5_10/PARTICLE_STILT.DAT does not contain any trajectory data.
```

**Comment by tartanrunner25** [comment]
Hello,
Good question, as I ran into this issue myself a few weeks ago. Long story short, I believe the issue is that read met_file subroutine does not account for forward based simulations, so in essence, the code grabs the wrong files, therefore causing the fortran code to complain with the above error. This routine would need to be modified. For my solution, I just concatenated the meteorological files together to avoid the issue. But in reality, the met_file subroutine needs to be modified. The postdoc that is working under me came up with a fix, but he's on vacation right now so I don't have the fix right this second. He should be back later this week so I can ask. I will probably need to update the repo with this, granted I probably won't be able to get around to this until the end of the month. I will check back when I hear back from colleague.

**Comment by BlueGrisGris** [comment]
Thanks for the reply, I just patched `find_met_files.r`  and made a pull request!
#100

**Comment by Liuc03** [comment]
Hi, I was wondering whether the problem had been resolved. Would you please share with me how to deal with calc_trajectory.r ?

**Comment by jmineau** [comment]
I ran into this issue awhile ago and made up my own fix, but never submitted a PR. I came to submit some other PR's and found this. I looked at my original fix and provided some updates: #102 . @benfasoli what are your thoughts?

**Comment by benfasoli** [comment]
Hey all 👋 I added some thoughts on #102.
